### PR TITLE
Maximum and Mode added

### PR DESCRIPTION
**Pull request by Protim-Proshun**
Maximum and mode added to the calculation, input_controller, view

### CODE REVIEW

File: metricstics/src/controller/input_controller.py [hunk 1]
```diff
@@ -42,6 +42,28 @@ def calculate_arithmetic_mean(self):
         mean.calculate_arithmetic_mean(self.data, result)
         return result
 
+        def view_maximum(self):
```
Comment (MajorTreble) [inline metricstics/src/controller/input_controller.py:45]: Why is called view? It should not be indented.

File: metricstics/src/controller/input_controller.py [hunk 2]
```diff
@@ -42,6 +42,28 @@ def calculate_arithmetic_mean(self):
...
+
+        Returns:
+            result(dict): The results of the calculations
+        """
+        result = {}
+        mean = Calculation()
+        mean.view_maximum(self.data, result)
+        return result
+
+    def view_mode(self):
```
Comment (MajorTreble) [inline metricstics/src/controller/input_controller.py:56]: Why is it called view?

File: metricstics/src/model/calculation.py [hunk 3]
```diff
@@ -42,6 +42,58 @@ def calculate_arithmetic_mean(self, data, result):
 
         result["ArithmeticMean"] = pymath.sum(data) / size
 
+        def view_maximum(self, data, result):
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:45]: Should not be indented or called view.

File: metricstics/src/model/calculation.py [hunk 4]
```diff
@@ -42,6 +42,58 @@ def calculate_arithmetic_mean(self, data, result):
...
+        """
+        Perform the Maximum.
+
+        Arg:
+            data(set): the data to work with
+            result(dic): store the result
+        """
+        data = data
+        maximum = data[0]
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:54]: What if size is 0?

File: metricstics/src/model/calculation.py [hunk 5]
```diff
@@ -42,6 +42,58 @@ def calculate_arithmetic_mean(self, data, result):
...
+            if i > maximum:
+                maximum = i
+        size = len(data)
+        if size == 0:
+            result["Maximum"] = 0
+            return
+
+        result["Maximum"] = maximum
+
+    def view_mode(self, data, result):
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:65]: Should not be called view

File: metricstics/src/model/calculation.py [hunk 6]
```diff
@@ -42,6 +42,58 @@ def calculate_arithmetic_mean(self, data, result):
...
+
+        Arg:
+            data(set): the data to work with
+            result(dic): store the result
+        """
+        data = data
+        max_count = 0
+        mode = 0
+
+        for i in data:
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:77]: What if size is 0?

File: metricstics/src/model/calculation.py [hunk 7]
```diff
@@ -42,6 +42,58 @@ def calculate_arithmetic_mean(self, data, result):
...
+                max_count = count
+                mode = current_number
+
+        size = len(data)
+        if size == 0:
+            result["Mode"] = 0
+            return
+
+        if max_count == 1:
+            result["Mode"] = 0
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:93]: Why 0?

File: metricstics/src/view/view.py [hunk 8]
```diff
@@ -34,13 +34,28 @@ def __init__(self, parent):
         # self.button_vmi.place(x=50, y=150)
         self.button_vmi.grid(row=3, column=1)
 
-        self.button_vmx = Button(self, text="View Maximum", borderless=1)
+        '''self.button_vmx = Button(self, text="View Maximum", borderless=1)
```
Comment (MajorTreble) [inline metricstics/src/view/view.py:37]: Don't submit commented code

**Review by MajorTreble — CHANGES_REQUESTED** [review]

File: metricstics/src/model/calculation.py (hunk 4 above)
Comment (Protim-Proshun) [reply metricstics/src/model/calculation.py:54]: it will return 0

File: metricstics/src/model/calculation.py (hunk 7 above)
Comment (Protim-Proshun) [reply metricstics/src/model/calculation.py:93]: if there is no reparation of any number the mode becomes "0"

File: metricstics/src/model/calculation.py (hunk 4 above)
Comment (MajorTreble) [reply metricstics/src/model/calculation.py:54]: if size is 0, you can return early without doing the for loop

File: metricstics/src/model/calculation.py (hunk 6 above)
Comment (MajorTreble) [reply metricstics/src/model/calculation.py:77]: same here, return early

File: metricstics/src/model/calculation.py (hunk 7 above)
Comment (MajorTreble) [reply metricstics/src/model/calculation.py:93]: An empty set does not have a mean so it is an error.

**Review by MajorTreble — APPROVED** [review]

File: metricstics/src/model/calculation.py (hunk 6 above)
Comment (Protim-Proshun) [reply metricstics/src/model/calculation.py:82]: Done

**Pull request merged**